[fix][download_data] Automatically generate dataset info in CLI help message

jidicula opened the pull request:
**Why this change was necessary**
The desired behaviour is to automatically generate a list of data
available using the URL dictionary.

**What this change does**
Modifies the format of the URL dictionary to include the dataset
description. Then, the URL dictionary is walked to generate a
multiline string containing the available dataset names and
descriptions. This string description is then used in the `help` kw
arg in the `click` command decorator for `main()`.

~~Note that I'm not sure what `prelude` does, so I have a TODO left in
the source code until I have more information.~~ Done

**Any side-effects?**
The spacing of the output is wonky as a result of the double
newline (\n) used. I tried using a single newline, but click seems to
get rid of those, which is *not* the behaviour of attempting to print
an f-string containing single newline chars:

```
print(f"hello\nhi")
```

This forced me to use double newline chars for each dataset option.

Furthermore, any attempt to indent each option doesn't seem to work
either, so I suspect that `click` does some kind of whitespace
trimming.

The resulting output of running the file with the `--help` flag:

```
$ python download_data.py --help
Usage: download_data.py [OPTIONS] DATA

  Download data from the internet. The available datasets are:

  - testing_data: Light-weighted dataset for testing purpose.

  - prelude: Binary for prelude software

Options:
  --verbose      Be more verbose.
  --output TEXT  Output folder.
  --help         Show this message and exit.
```

**Additional context/notes/links**

Related to #91 - Address TODOs in cli/download_data.py

jidicula committed:
[fix][download_data] Fix CLI help message

**Why this change was necessary**
The desired behaviour is to automatically generate a list of data
available using the URL dictionary.

**What this change does**
Modifies the format of the URL dictionary to include the dataset
description. Then, the URL dictionary is walked to generate a
multiline string containing the available dataset names and
descriptions. This string description is then used in the `help` kw
arg in the `click` command decorator for `main()`.

Note that I'm not sure what `prelude` does, so I have a TODO left in
the source code until I have more information.

**Any side-effects?**
The spacing of the output is wonky as a result of the double
newline (\n) used. I tried using a single newline, but click seems to
get rid of those, which is *not* the behaviour of attempting to print
an f-string containing single newline chars:

```
print(f"hello\nhi")
```

This forced me to use double newline chars for each dataset option.

Furthermore, any attempt to indent each option doesn't seem to work
either, so I suspect that `click` does some kind of whitespace
trimming.

The resulting output of running the file with the `--help` flag:

```
$ python download_data.py --help
Usage: download_data.py [OPTIONS] DATA

  Download data from the internet. The available datasets are:

  - testing_data: Light-weighted dataset for testing purpose.

  - prelude: Binary for prelude software

Options:
  --verbose      Be more verbose.
  --output TEXT  Output folder.
  --help         Show this message and exit.
```

**Additional context/notes/links**

Related to #91 - Address TODOs in cli/download_data.py

jidicula committed:
[fix][download_data] Remove TODO about prelude

**Why this change was necessary**
Got input from Julien about what prelude does.

**What this change does**
See title

**Any side-effects?**
None

**Additional context/notes/links**

Resolves #91 - Address TODOs in cli/download_data.py

jidicula committed:
[fix][download_data] Add correct subscripting for URL dict

**Why this change was necessary**
The URL dict format was changed, so accessing the URL requires 1 more
level of subscripting of an item in the dict.

**What this change does**
See title

**Any side-effects?**
None

**Additional context/notes/links**

Resolves #91 - Address TODOs in cli/download_data.py

po09i reviewed (commented):
Aside from those two PEP8 compatibility issue, everything else seems good. I'll approeve after.

jidicula committed:
[style][download_data] Add linebreaks for PEP8 compatibility

**Why this change was necessary**
Lines were noncompliant with PEP8 for line length and newlines before
function definition.

**What this change does**
See title.

**Any side-effects?**
None

**Additional context/notes/links**

Resolves
https://github.com/shimming-toolbox/shimming-toolbox-py/pull/93#discussion_r491476407
and
https://github.com/shimming-toolbox/shimming-toolbox-py/pull/93#discussion_r491476465
